packages/js-client/src/context.ts

```diff
@@ -0,0 +1,71 @@
+import { SimpleStorageContextParams } from './types';
+import { Context, ContextCore, ContextState } from '@aragon/sdk-client-common';
+
+// TODO
+// import from @aragon/sdk-client-common
+type OverriddenState = {
+  daoFactoryAddress: boolean;
+  ensRegistryAddress: boolean;
+  gasFeeEstimationFactor: boolean;
+  ipfsNodes: boolean;
+  graphqlNodes: boolean;
+};
+
+type SimpleStorageContextState = ContextState & {
+  // extend the Context state with a new state for storing
+  // the new parameters
+  myParam: string;
+};
+
+type SimpleStorageOverridenState = OverriddenState & {
+  myParam: boolean;
+};
+
+export class SimpleStorageContext extends ContextCore {
+  // super is called before the properties are initialized
+  // so we initialize them to the value of the parent class
+  protected state: SimpleStorageContextState = this.state;
+  protected overriden: SimpleStorageOverridenState = this.overriden;
+  constructor(
+    contextParams?: Partial<SimpleStorageContextParams>,
+    context?: Context
+  ) {
+    // parent contructor will call this.set(contextParams)
+    // so we don't need to call it again
+    super(contextParams);
+    if (context) {
+      // copy the context properties to this
+      Object.assign(this, context);
+    }
+    if (contextParams) {
+      // overide the context params with the ones passed to the constructor
+      this.set(contextParams ?? {});
+    }
+  }
+
+  public set(contextParams: SimpleStorageContextParams) {
+    // the super function will call this set
+    // so we need to call the parent set first
+    super.set(contextParams);
+    // set the default values for the new params
+    this.setDefaults();
+    // override default params if specified in the context
+    if (contextParams.myParam) {
+      // override the myParam value
+      this.state.myParam = contextParams.myParam;
+      // set the overriden flag to true in case set is called again
+      this.overriden.myParam = true;
+    }
+  }
```

Comment:
Shoudn't this be consistent with the constructor? or at least detect which field is handled by whom?